### ISSUE TITLE
Building is too slow

### ISSUE THREAD

**Issue by sensed** [Microsoft/VSLinux#93]
Hello,

Basically that plugin is exactly what I looked for: Visual Studio convenience combined with Linux development. 
But for more extensive projects this concept of building is too slow. It compiles each single code file with a separate compiler call over ssh. 
It would be smarter and much faster to generate a makefile, copy it to target device and call make. So it would also be able to use multi-processor building (with -j parameter).
Another IDE i'm currently using which supports this remote building does it like this way and compiles 10 times faster (same code, same platform, same target). So i have to resign from "VS C++ for linux", I vainly hoped it would be the cure.
Perhaps also zipping the source files will reduce the time needed to copy them to target device.
Also the debugger is not stable, but I think, that is a common GDB problem. Other IDEs have similar problems.

There is still air up. Keep it up!

**Comment by AliAskari** [comment]
@sensed In my experience the extension is much more stable in the VS 2017 RC. You even have the option to attach to the remote process. But the attach part doesn't work all the time. I also suggest to try it in both gdb and gdbserver mode. Surprisingly I have found that when I debug in gdbserver mode it is more stable. 

Do you know that you can use the makefile project in this extension? I have setup a SAMBA share where I have everything as a CMake project on my target. The only extra step is to add your custom build/rebuild, with -j to have multi core builds, and clean commands in the project settings. It is as fast as building natively.

**Comment by sensed** [comment]
Yes. I know Makefile project is also possible, but i was porting a extensive visual studio solution to Linux and therefore i did just use this solution and added the Linux Toolset. So i have one VS Project for multiple Platforms. To rebuild this 200+-sourcefile-project as Makefile-project with multiple Platforms (Win32, Win IoT, Linux, VxVorks)... is too time-consuming. And CMake does also have no self-explanatory programming syntax. But it seems, there is no further way. 
NetBeans as example is doing this job with bravery.

**Comment by ohanson511** [comment]
@sensed adding the linux toolset to an existing vcprojx and making it multiplatform sounds extremely interesting.  Do you mind sharing with us how you did this ? From current documentation I only know how to create a new linux project and would love to implement your solution instead.

**Comment by itodirel** [comment]
we just currently implemented parallel builds support, we are also working on connection caching, with those two, the performance would be close to that of local Windows

**Comment by itodirel** [comment]
please feel free to activate if you have more questions

**Comment by levhaikin** [comment]
what version of visual-studio should start supporting this great feature?

**Comment by margaretharp** [comment]
I am using Visual Studio 2017 version 15.8.7 with the Linux workload on a Windows 10 (Version 1703) PC.  My sources are in a C: folder which is mapped to a mount point on my VMWare Linux virtual machine, so I've edited CMakeSettings.json to not copy the sources to the remote machine.  I downloaded the required MS CMake 3.8 (which supports server).  Opening the root folder which contains the base CMakeList.text of my very very large project, which triggers the parsing of all of the nested CMakeList.txt files results in an eternal wait while the CMake Targets view shows <root folder>"/CMakeLists.text is still parsing".  I let this run overnight and it never finished.  I closed Visual Studio, deleted all of the .vs folders, and opened my root folder again.  After it finished the initial parse, it still wasn't display my CMake Targets.  Only after I selected "Generate CMake Cache" from the Cmake toolbar menu did it complete, 80 minutes after I had reopened the root folder.  
Typically I can't tell whether cmake is hung again, just very very slow, or I need to exit Visual Studio, clean up some detritus and restart it.  Any clues?
